internal/deviations/deviations.go

```diff
@@ -99,4 +108,21 @@ var (
 	StaticProtocolName = flag.String("deviation_static_protocol_name", "DEFAULT", "The name used for the static routing protocol.  The default name in OpenConfig is \"DEFAULT\" but some devices use other names.")
 
 	DeprecatedVlanID = flag.Bool("deviation_deprecated_vlan_id", false, "Device requires using the deprecated openconfig-vlan:vlan/config/vlan-id or openconfig-vlan:vlan/state/vlan-id leaves.")
+
+	AddSubIntfToNetInst = flag.Bool("deviation_adding_subintf_to_netowork_instance", true,
+	"Vendor requiring explicitly attaching sub-interface to network-instance, this deviation should be marked true. Default expectation is sub-interface is part of default network-instance implicitly.")
 )
+
+func SubIntfToNetworkInstance(t *testing.T, dconf *ondatra.Config, i *telemetry.Interface, si uint32, inst string) {
```

Comment:
There are two implementations we could have of this -- we could handle it at the end of the configuration generation - i.e., go and make sure that every interface that is configured is associated with a network instance, and if not, then make that association -- or alternatively this design. Did you consider the alternate approach in the wider implementation?

Reply:
Resolving conversation as this was moved to AddToDUT() in `attrs.go`